lib/auth/auth.go

```diff
@@ -5204,6 +5210,16 @@ func (a *Server) CreateAccessRequestV2(ctx context.Context, req types.AccessRequ
 	}
 	req.SetRequestedResourceIDs(requestedResourceIDs)
 
+	err = okta.CheckResourcesRequestable(ctx, requestedResourceIDs, okta.AccessPoint{
+		Plugins:              a.Plugins,
+		UnifiedResourceCache: a.UnifiedResourceCache,
+	})
+	if errors.Is(err, okta.OktaResourceNotRequestableError) {
+		return nil, trace.Wrap(err)
+	} else if err != nil {
+		return nil, trace.Wrap(err, "checking if Okta-originated resources are requestable")
+	}
```

Comment:
nit: move to a separate function to hide error complexity handling. 

Also wonder if we can move it to  `ServerWithRoles-CreateAccessRequestV2 a.authServer.CreateAccessRequestV2->Server.CreateAccessRequestV2->Validate`

where most of validation of access request function are evaluated

Reply:
Can't be moved because of the `a.appendImplicitlyRequiredResources` call above